File: Happy Number.cpp
class Solution {
public:

        int num(int n){
            long long x=0;
        while (n!=0){
            int temp= n%10;
            x+=temp*temp;
            n=n/10; 
        }
        return x;
        }
    bool isHappy(int n) {
        unordered_set<int>set;
        while(n!=1 && !set.count(n)){
            set.insert(n);
            n=num(n);
        }
        return n==1;
    }
};

File: week-2/python/happy-number.py
class Solution:
    def num(self, n):
        x = 0
        while n != 0:
            temp = n % 10
            x += temp * temp
            n = n // 10
        return x
        
    def isHappy(self, n: int) -> bool:
        nums_set = set()
        while n != 1 and n not in nums_set:
            nums_set.add(n)
            n = self.num(n)
        return n == 1